recipes/GigaSpeech/gigaspeech_prepare.py

```diff
@@ -0,0 +1,693 @@
+        https://huggingface.co/datasets/speechcolab/gigaspeech
+        The dataset will be downloaded in the default folder specified in the
+        environment variable HF_HUB_CACHE. Please change it if necessary.
+    punctuation : bool, optional
+        Keeping the punctuation, or not.
+    filler : bool, optional
+        Keeping filler words (hum), or not.
+    hf_multiprocess_load: bool, optional
+        If True, all the CPU threads will be used for data prepration. If set to
+        False, only one will be. Note that the data prepration of the larger sets
+        on a single core car take more than 24 hours (from downloading to done).
+
+    Returns
+    -------
+    None
+    """
+    logger.info(f"Preparing GigaSpeech dataset in {save_folder}...")
+
+    if skip_prep:
+        logger.info("Skipping data preparation as `skip_prep` is set to `True`")
+        return
+
+    # check that `splits` input is valid
+    for split in splits:
+        assert (
+            split in SPLITS + TRAIN_SUBSET
+        ), f"Split {split} not recognized. Valid splits are {SPLITS + TRAIN_SUBSET}."
+
+    # check that we are not using multiple train subsets
+    if len(set(splits).intersection(TRAIN_SUBSET)) > 1:
+        raise ValueError(
+            "You cannot use multiple train subsets. Please select only one train subset."
+        )
+
+    os.makedirs(save_folder, exist_ok=True)
+
+    # Setting output paths
+    save_output = {}
+    train_split = ""
+    for split in splits:
+        if split in TRAIN_SUBSET:
+            save_output["train"] = output_train
+            train_split = split
+        else:
+            if split == "DEV":
+                save_output["validation"] = output_dev
+            elif split == "TEST":
+                save_output["test"] = output_test
+
+    # check if the data is already prepared
+    if skip_csv(save_output):
+        logger.info("Skipping preparation, completed in previous run.")
+        return
+    else:
+        logger.info("Starting data preparation...")
+
+    if download_with_HF:
+        from datasets import load_dataset
+
+        if os.path.exists("dataset.py"):
+            logger.info("HuggingFace dataset.py found.")
+        else:
+            raise FileNotFoundError(
+                "HuggingFace dataset.py not found. Please run this recipe from the correct recipe folder or copy the dataset.py file."
+            )
+
+        if "HF_HUB_CACHE" in os.environ:
+            hf_caching_dir = os.environ["HF_HUB_CACHE"]
+        else:
+            hf_caching_dir = os.environ["HF_HOME"]
+
+        logger.info(
+            "Downloading dataset from HuggingFace to: " + str(hf_caching_dir)
+        )
+        logger.info(
+            "To change this directory modify the HF_HUB_CACHE env. variable."
+        )
+
+        nproc = 1
+        if hf_multiprocess_load:
+            import multiprocessing
+
+            nproc = multiprocessing.cpu_count()
```

Comment:
isn't `nproc` supposed to be `num_workers` ?

Reply:
No. nproc here is independent to torch. It's the actual number of available cores, to make it faster.